src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -136,6 +144,12 @@ public AddressBook load() throws StorageOperationException {
         // create empty file if not found
         } catch (FileNotFoundException fnfe) {
             final AddressBook empty = new AddressBook();
+            // XXX hack to create an empty file since we now do not allow non existent files
+            try {
+                Files.createFile(path);
+            } catch (IOException e) {
+                throw new StorageOperationException("Cannot create file!");
+            }
```

Comment:
while I like your idea of creating a file when the file is not found, it still makes sense for the program to throw an exception. For example, the program might want to tell the user that the previous file is not found, thus need to create a new file. Of course we can do it by normal return boolean or string, a more neat way in OOP is to throw an exception to tell the one who is calling the function: hey, the file is missing but I recover  it, can you do something about this? Then the caller would say ok, I will ask the UI to display the message. ^^